Tevling/Services/ChallengeService.cs

```diff
@@ -305,4 +306,71 @@ public async Task<ScoreBoard> GetScoreBoardAsync(int challengeId, CancellationTo
 
         return new ScoreBoard(scores);
     }
+
+    public async Task<Athlete?> DrawChallengeWinnerAsync(int challengeId, CancellationToken ct = default)
+    {
+        using DataContext dataContext = await _dataContextFactory.CreateDbContextAsync(ct);
+
+        Challenge? challenge = await dataContext.Challenges
+            .Include(c => c.Athletes)
+            .ThenInclude(a => a.Activities)
```

Comment:
I'm a bit worried that lifting all activities for all athletes in the challenge will consume a lot of memory... More and more so as the activity table grows. Is there a way to limit the activities we include/fetch to the time span of the challenge? 🤔

Reply:
Yeah, it might not be supported to do it directly, at least not with the SQLite provider, so perhaps we'll have to split it up into two queries.. Don't bother with it now, I'll have a look at it later 🙂